### PR TITLE
Do not continue API logic after an error detection

### PR DESCRIPTION
**Pull request by hevrard**
Once an error has been detected, we should not try to keep
executing the rest of the subroutine.

### CODE REVIEW

File: gapis/api/vulkan/extensions/khr_descriptor_update_template.api [hunk 1]
```diff
@@ -124,7 +123,10 @@ sub ref!DescriptorUpdateTemplateObject createDescriptorUpdateTemplate(
     if !(device in Devices) { vkErrorInvalidDevice(device) }
     if pCreateInfo == null { vkErrorNullPointer("pDescriptorUpdateTemplate") }
     info := pCreateInfo[0]
-    if !(info.descriptorSetLayout in DescriptorSetLayouts) { vkErrorInvalidDescriptorSetLayout(info.descriptorSetLayout) }
+    if !(info.descriptorSetLayout in DescriptorSetLayouts) {
+        vkErrorInvalidDescriptorSetLayout(info.descriptorSetLayout)
+        abort
```
Comment (hevrard) [inline gapis/api/vulkan/extensions/khr_descriptor_update_template.api]: I am not sure how to handle the error here, as a return value is expected? Should we return nil, and if so how do we express this?

Comment (hevrard) [reply gapis/api/vulkan/extensions/khr_descriptor_update_template.api]: @AWoloszyn a friendly ping :)

Comment (AWoloszyn) [reply gapis/api/vulkan/extensions/khr_descriptor_update_template.api]: You can make a nil object, but it is a bit awkward.

```
class NilDescriptorUpdateTemplateObject {
    ref!DescriptorUpdateTemplateObject v;
}

....
....
obj := NilDescriptorUpdateTemplateObject().v
if (info.descriptorSetLayout in DescriptorSetLayouts) {
  obj = .....
}
return obj
```

Comment (hevrard) [reply gapis/api/vulkan/extensions/khr_descriptor_update_template.api]: Thanks, I ended up opting for:
```
...
obj := new!DescriptorUpdateTemplateObject()
if (check error) {
  // report error
} else {
  obj.Device = device
  // etc...
}
return obj
```

File: gapis/api/vulkan/extensions/khr_descriptor_update_template.api [hunk 2]
```diff
@@ -122,102 +121,110 @@ sub ref!DescriptorUpdateTemplateObject createDescriptorUpdateTemplate(
...
-        FromKHR: fromKHR,
-    )
-
-    obj.Entries = rewriteDescriptorUpdateEntries(
-        info.descriptorUpdateEntryCount,
-        info.pDescriptorUpdateEntries,
-        info.descriptorSetLayout)
-
+    obj := new!DescriptorUpdateTemplateObject()
+    if !(device in Devices) {
```
Comment (AWoloszyn) [inline gapis/api/vulkan/extensions/khr_descriptor_update_template.api]: I would switch the device section back, so it matches every other command.

Comment (hevrard) [reply gapis/api/vulkan/extensions/khr_descriptor_update_template.api]: Done

**Pull request merged**